tests/unit/versioning/VersionID.spec.js

```diff
@@ -9,83 +10,124 @@ function randkey(length) {
+        it.skip('should return error decoding non-hex string versionIds', () => {
+            const encoded = vids.map(vid => VID.hexEncode(vid));
+            const decoded = encoded.map(vid => VID.hexDecode(`${vid}foo`));
+            decoded.forEach(result => assert(result instanceof Error));
+        });
 
-    it('should encode and decode versionIds', () => {
-        const encoded = vids.map(vid => VID.hexEncode(vid));
-        const decoded = encoded.map(vid => VID.hexDecode(vid));
-        assert.strictEqual(vids.length, count);
-        assert.deepStrictEqual(vids, decoded);
-    });
+        it('should encode and decode versionIds', () => {
+            const encoded = vids.map(vid => VID.hexEncode(vid));
+            const decoded = encoded.map(vid => VID.hexDecode(vid));
+            assert.strictEqual(vids.length, count);
+            assert.deepStrictEqual(vids, decoded);
+        });
 
-    it('simple base62 version test', () => {
-        const vid = '98376906954349999999RG001  145.20.5';
-        const encoded = VID.base62Encode(vid);
-        assert.strictEqual(encoded, 'aJLWKz4Ko9IjBBgXKj5KQT2G9UHv0g7P');
-        const decoded = VID.base62Decode(encoded);
-        assert.strictEqual(vid, decoded);
-    });
+        it('should encode and decode correctly with legacy format', () => {
+            const encoded = vids.map(vid => VID.encode(vid));
+            const decoded = encoded.map(vid => VID.decode(vid));
 
-    it('base62 version test with smaller part1 number', () => {
-        const vid = '00000000054349999999RG001  145.20.5';
-        const encoded = VID.base62Encode(vid);
-        const decoded = VID.base62Decode(encoded);
-        assert.strictEqual(vid, decoded);
+            assert.strictEqual(vids.every(x => x.length > 27), true);
+            assert.strictEqual(encoded.every(x => x.length > 32), true);
+            assert.deepStrictEqual(vids, decoded);
+        });
     });
 
-    it('base62 version test with smaller part2 number', () => {
-        const vid = '98376906950000099999RG001  145.20.5';
-        const encoded = VID.base62Encode(vid);
-        const decoded = VID.base62Decode(encoded);
-        assert.strictEqual(vid, decoded);
-    });
 
-    it('base62 version test with smaller part3', () => {
-        const vid = '98376906950000099999R1  145.20.5';
-        const encoded = VID.base62Encode(vid);
-        const decoded = VID.base62Decode(encoded);
-        assert.strictEqual(vid, decoded);
-    });
+    describe('Short IDs', () => {
+        env.S3_VERSION_ID_ENCODING_TYPE = 'base62';
+        const vids = generateRandomVIDs(count);
 
-    it('base62 version test with smaller part3 - 2', () => {
-        const vid = '98376906950000099999R1x';
-        const encoded = VID.base62Encode(vid);
-        const decoded = VID.base62Decode(encoded);
-        assert.strictEqual(vid, decoded);
-    });
+        it('sorted in reversed chronological and alphabetical order', () => {
```

Comment:
Woulnd't this test better be done outside of the encoding scheme `describe` blocks, only once? Because I don't think it depends on the encoding scheme, it's just checking that internal version IDs are properly ordered without encoding/decoding anything.

Reply:
I think its better to explicitly show that the new format won't break ordering in the spec

Reply:
its only obvious to us that ordering is preserved because we know exactly what its doing (just chopping off the info field if there is one) in the new format.

Reply:
Yes, it makes sense since it is removing the info, I did not get this at first.